packages/api/tests/endpoints/test-rules.js

```diff
@@ -390,16 +514,77 @@ test.serial('POST returns a 500 response if record creation throws unexpected er
   }
 });
 
+test.serial('POST does not write to RDS or DynamoDB if writing to RDS fails', async (t) => {
+  const { newRule, dbClient } = t.context;
+
+  const failingDbClient = {
+    transaction: () => {
+      throw new Error('Insert Rule Error');
+    },
+  };
```

Comment:
This mock actually ensures that **neither the Dynamo or RDS write can happen**, since they both happen within a `knex.transaction`. I think you need a mock that looks more like this:

https://github.com/nasa/cumulus/blob/feature-rds/packages/api/tests/lambdas/sf-event-sqs-to-db-records/test-write-execution.js#L530-L539

Reply:
[`9e37ed6` (#2012)](https://github.com/nasa/cumulus/pull/2012/commits/9e37ed6f9ae36d213d879fe88fcaab071d440c51)